pkg/generate/app/cmd/newapp.go

```diff
@@ -222,27 +224,58 @@ func (c *AppConfig) SetOpenShiftClient(osclient client.Interface, OriginNamespac
 // AddArguments converts command line arguments into the appropriate bucket based on what they look like
 func (c *AppConfig) AddArguments(args []string) []string {
 	unknown := []string{}
+	c.ArgumentClassificationErrors = map[string]error{}
 	for _, s := range args {
-		switch {
-		case cmdutil.IsEnvironmentArgument(s):
+		if len(s) == 0 {
+			continue
+		}
+
+		validated := false
+
+		// had to move off of elegant switch to capture errors for future display, but mimic
+		// order of precedence established by switch on env arg, then source repo, then comp
+		// ref, then possible template, with no fallthroughs and updates to c.Components or
+		// c.SourceRepositories if a classification of higher precedence succeeds
+
+		rc := cmdutil.IsEnvironmentArgument(s)
+		if rc {
 			glog.V(2).Infof("treating %s as possible environment argument\n", s)
 			c.Environment = append(c.Environment, s)
-		case app.IsPossibleSourceRepository(s):
+			validated = true
+		}
+
+		remote, err := app.IsRemoteRepository(s)
+		c.ArgumentClassificationErrors["Remote access"] = err
+		local, err := app.IsDirectory(s)
+		c.ArgumentClassificationErrors["Local access"] = err
```

Comment:
> And on classificatnoi errors and err != nil, again, that was intentional. If a classification passed, I wanted it illustrated by the fact that the error field was nil. If I don't add a key / nil err pair, then unless you know the entire list of classifications, you don't know what has passed if we only list the actual errors.

yeah i sort of figured that was the intention, but for an end user it's a pretty ugly/confusing output.  I *might* be convinced of keeping the nil value and doing better processing when printing the errors out to reformat the "nil" to something more informative to the user, but overall I think doing anything with it is tipping too far on the side of "what gabe+ben want to see to debug the problem" and not "what is useful to a user to see to know what they should do to fix it".  For example "Component reference" means nothing at all to an end user.